spring-boot-modules/spring-boot-validations/src/test/java/com/baeldung/dto/BooleanUnitTest.java

```diff
@@ -0,0 +1,28 @@
+package com.baeldung.dto;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+class BooleanUnitTest {
+
+    @Test
+    void givenInputAsString_whenStringToBoolean_thenValidBooleanConversion() {
+        Boolean trueVar = Boolean.valueOf("TRUE");
+        Boolean falseVar = Boolean.valueOf("false");
+        Boolean parsedVar = Boolean.parseBoolean("True");
+
+        assertEquals(Boolean.TRUE, trueVar);
+        assertEquals(Boolean.FALSE, falseVar);
+        assertEquals(Boolean.TRUE, parsedVar);
+    }
+
+    @Test
+    void givenInputAsboolean_whenbooleanToBoolean_thenValidBooleanConversion() {
+        Boolean trueVar = Boolean.valueOf(true);
+        Boolean falseVar = Boolean.valueOf(false);
+
+        assertEquals(Boolean.TRUE, trueVar);
+        assertEquals(Boolean.FALSE, falseVar);
+    }
```

Comment:
For these two tests, instead of assigning variables and then asserting on their values, just put the expression being tested in the assertEquals calls. For example:
`assertEquals(Boolean.TRUE, Boolean.valueOf("True");`

Reply:
Updated as per review comment